compiler-core/src/language_server/signature_help.rs

```diff
@@ -0,0 +1,241 @@
+                        .field_map(),
+                };
+                let name = format!("{module_alias}.{label}").into();
+                signature_help(name, fun, args, field_map)
+            }
+
+            // In all other cases we can't figure out a good name to show in the
+            // signature help so we use an anonymous `fn` as the name to be
+            // shown.
+            //
+            //     eg. fn(a){a}(|)
+            //                  ^ When the cursor is here we are going to show
+            //                    "fn(a: a) -> a" as the help signature.
+            //
+            _ => signature_help("fn".into(), fun, args, None),
+        },
+
+        _ => None,
+    }
+}
+
+/// Show the signature help of a function with the given name.
+/// Besides the function's typed expression `fun`, this function needs a bit of
+/// additional data to properly display a useful help signature:
+///
+/// - `fun_name` is used as the display name of the function in the help
+///   signature.
+/// - `supplied_args` are arguments being passed to the function call, those
+///   might not be of the correct arity or have wrong types but are used to
+///   deduce which argument should be highlighted next in the help signature.
+/// - `field_map` is the function's field map (if any) that will be used to
+///   display labels and understand which labelled argument should be
+///   highlighted next in the help signature.
+///
+fn signature_help(
+    fun_name: EcoString,
+    fun: &TypedExpr,
+    supplied_args: &[CallArg<TypedExpr>],
+    field_map: Option<&FieldMap>,
+) -> Option<SignatureHelp> {
+    let (args, return_) = fun.type_().fn_types()?;
+
+    // If the function has no arguments, we don't want to show any help.
+    let args_len = args.len() as u32;
+    if args_len == 0 {
+        return None;
+    }
+
+    let index_to_label = match field_map {
+        Some(field_map) => field_map
+            .fields
+            .iter()
+            .map(|(name, index)| (*index, name.clone()))
+            .collect(),
+        None => HashMap::new(),
+    };
+
+    let printer = Printer::new();
+    let (label, parameters) =
+        print_signature_help(printer, fun_name, args, return_, &index_to_label);
+
+    let active_parameter = active_parameter_index(supplied_args, index_to_label)
+        // If we don't want to highlight any arg in the suggestion we have to
+        // explicitly provide an out of bound index.
+        .or(Some(args_len));
+
+    Some(SignatureHelp {
+        signatures: vec![SignatureInformation {
+            label,
+            documentation: fun.get_documentation().map(|d| {
+                Documentation::MarkupContent(MarkupContent {
+                    kind: MarkupKind::Markdown,
+                    value: d.into(),
+                })
+            }),
+            parameters: Some(parameters),
+            active_parameter: None,
+        }],
+        active_signature: Some(0),
```

Comment:
What does this mean?

Reply:
This is the index of the signature to use from the list of signatures we include in the response. Since I always include just a single signature I'm telling the LS to use that